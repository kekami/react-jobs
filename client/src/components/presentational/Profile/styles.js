import styled from 'styled-components';

/* Start of General Profile styling */
export const ProfileWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding-top: 10px;
    padding-bottom: 10px;
    background-color: #DDEDF5;
`;
export const ProfileContainer = styled.div`
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    background-color: #fff;
    border-radius: 25px;
    margin: auto;
    min-width: 100vw;
    margin-top: 4em;
    margin-bottom: 2em;

    @media (min-width: 500px) {   
        min-width: 90vw;
        max-width: 90vw;
        padding: 0em 1.5em 1.5em 1.5em ;
    }

    @media (min-width: 650px) {   
        min-width: 80vw;
        max-width: 80vw;
        padding: 0em 2em 2em 2em ;
    }

    @media (min-width: 800px) {   
        min-width: 70vw;
        max-width: 70vw;
        padding: 0em 3em 3em 3em ;

    }
`;
/* End of General Profile styling */
/* Start of Profile Navigation styling */
export const NavWrapper = styled.div`

    nav {
        ul {
            display: flex;
            flex-direction: row;
            justify-content: center;
            align-items: center;
            padding: 0;

            li {
                display: block;
                text-align: center;
                font-size: 0.9em;
                width: 50%;
                padding: 0;
                
                a {
                    display: block;
                    text-decoration: none;
                    padding: 1em 0em;
                }
                a:first-of-type {
                    color: #1c75bc;
                }

            }
        }
    }
`;
/* End of Profile Navigatoin styling */

/* Start of the upper part of the Profile styling */
export const UpperSectionWrapper = styled.div`
    maring: auto;
    border-bottom: none;
    @media (min-width: 650px) { 
        margin: 0.7em; 
        padding: 1em;
    }
    @media (min-width: 1000px) {
        margin: 1em;
        padding: 1.2em;
    }
    header {
        font-size: 1.2em;
        padding: 2em 0em 2em 0em;
        text-align: center;
        color: #04d092;

        strong {
            color: #9b9b9b;
        }

    }
    div {
        text-align: center;
    }
    p {
        text-align: center;
        line-height: 2em;
        font-size: 0.9em;

        @media(min-width:1000px) {
            padding: 1em 2.5em 0em 2.5em;
        }
        
    }
`;
export const PicWrapper = styled.div`
    display: flex;
    flex-direction: row;
    margin:auto;
    justify-content: center;

`;
export const Avatar = styled.div`
    border-radius:50%;
    background-color: white;
    border: dashed 2px black;
    width: 160px;
    height: 160px;
    padding: 20px;
    z-index:4;

    @media(min-width:1000px){
        margin-top: -140px;
        margin-bottom: -70px;
    }
`;
export const PicUploader = styled.div`
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    border: 2px dashed #04d092;
    border-radius: 10px;
    justify-content: center;

    div {

        padding: 1.4em 0.2em 1.4em 0.2em;
        line-height: 1.7em;
        margin: auto;
        display: flex;
        font-size: 0.85em;
        position: relative;


        
        input {        
            display:inline-block;
            position: absolute;
            opacity: 0;
            cursor: pointer;
            z-index: 1;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            align-items: baseline;
            text-align: start;
                
        }


        @media(min-width:800px){
            z-index:0;
            max-width: 65%;
           
        }
        
    }
`;

export const ProgressSection = styled.div`
    display: flex;
    flex-flow: column wrap;
    justify-items: center;
`;

export const ProgressWrapper = styled.div`
    width: 100%;
    background-color: #f0b14a;
    border-radius: 8px  
`;
export const ProgressBar = styled.div`
    height: 8px;
    background-color: #04d092;
    border-radius: 8px
    ;
`;
/* End of the upper part of the Profile styling */
/* Start of General Section styling */
export const SectionWrapper = styled.div`
    background-color: ${props => (props.edited ? '#FAFAFA' : ' white')};
    border: ${props => (props.edited ? 'solid 1px #DDEDF5' : 'none')};
    border-bottom: solid 1px ${props => (!props.edited ? '#DDEDF5' : 'none')};
    marging: auto;
    form {
        padding: 1em;
        
        p {
            color: grey;
        }
        button {
            -webkit-transition-duration: 0.4s;
            transition-duration: 0.4s;
            cursor: pointer;
            background-color: #04d092;
            border: none;
            color: white;
            padding: 15px 32px;
            text-align: center;
            text-decoration: none;
            display: inline-block;
            font-size: 16px;

            &:hover {
                background-color: #9b9b9b;
            }
        }
    }

    @media (min-width: 650px) { 
        margin: 0.7em; 
        padding: 1em;

        form {
            p {
                padding: 0.2em;
            }
        }
    }

    @media (min-width: 1000px) {
        margin: 1em;
        padding: 1.2em;
    }
`;
export const SectionHead = styled.div`
    padding: 1em;  
`;
export const Btn = styled.button` 
    border: none;
    outline:none;
    background-color: #eee;
`;
export const Edit = styled.button`
    padding: 0.4em;
    background-color: inherit;
    border: none;
    cursor: pointer;
    color: #04d092;
    outline:none;
`;
export const I = styled.i`
    padding-right: 0.2em;
`;
/* End of General Section styling */

/* Start of Skills and Languages Sections styling */
export const First2 = styled.div`
 
`;
export const SkillHead = styled.div`
    display:flex;
    flex-direction: row;
    justify-content: space-between;
    padding-left: 1em;
    padding-right: 1em;
    padding-bottom: 1em;
    color: #9b9b9b;
    
    strong {
        color: black;
        font-size: 1.3em;
        padding-right: 0.9em;

    }
    
`;
export const LangHead = styled(SkillHead)``;
export const LinkHead = styled(SkillHead)``;
export const Link = styled.a``;
export const TagsForm = styled.div`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    padding-left: 1em;
`;
export const SkillTag = styled.p`
    display: ${props => (props.showing ? 'inline-block' : 'none')};
    background-color: #eee;
    color: black;
    padding: 10px;
    margin: 5px 10px 5px 0px;
    font-size: 0.8rem;
    border: solid 2px #EEE;
    border-radius: 5px;
`;
export const LangTag = styled(SkillTag)``;

/* Start of Links Section styling */
export const Inputform = styled.form`
    display: flex;
    flex-direction: column;
    input {
        width: 100%;
        margin: 0;
        padding: 0.5em;
        
    }

    @media (min-width: 800px) {
        flex-direction: row;
        justify-content: left;

        input {
            margin: 0;
            width: 25em;
        }
    }
`;

export const SocMediaInput = styled.div`
    display: flex;
    flex-flow: column wrap;

    
    div {
        -webkit-box-sizing: border-box;
        -moz-box-sizing: border-box;
        box-sizing: border-box;
        display: flex;
        flex-flow: row wrap;
        padding: 0.1em;
        margin-top: 0.8em;
        margin-bottom: 0.8em;
        justify-content: space-between;
        align-items: center;
       

        label {
            min-width: 25%;
            padding: 0.2em;
        }

        input {
            margin-right: 5em;
            padding: 0.5em;
            min-width: 5em;
            
        }
    }
    

    @media (min-width: 1000px) {
        input {
            margin: 5;
            width: 25em;
        }
    }


`;
export const Label = styled.label`
    min-width:15em;
    align-self: center;
`;
export const Websites = styled.div`
    margin-top: 1em;
    margin-bottom: 1em;
`;
export const LinkTag = styled.div`
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;

    display:flex;
    flex-flow: row wrap;   
    align-items: center;
    padding: 0.2em;
    margin-top: 0.5em;
    margin-bottom: 0.5em;

    div {
        display: flex;
        padding: 0.2em;
        font-size: 1em;
        min-height: 1.5em;
        min-width:1.5em;
        margin: 0.5em;

        i {
            font-size: 1em;
        }
        
    }

    p {
        margin-left: 1em;
        a {
            text-decoration: none;
        }
    }

    @media (min-width: 700px) {
        div {
            display: flex;
            font-size: 1.5em;
            min-height: 2em;
            min-width:2em;
            
    
            i {
                
                font-size: 1.5em;
            }
            
        }
        p {
            margin-left: 2em;
            min-width: 4em;
        }
    }

`;
/* Start of Experience Section styling */

export const ExperienceHead = styled(LinkHead)``;
export const Click2AddWork = styled.div`
    -webkit-transition-duration: 0.6s;
    transition-duration: 0.6s;
    width: inherit;
    cursor: pointer;
    padding: 1em;
    background-color: #04d092;
    color: white;

    I {
        color: white;
        padding: 0.2em;
    }

    &:hover {
        background-color: white;
        color: #04d092;

        I {
            color: #04d092;
            padding: 0.2em;
        }
    }

`;
export const ExperienceInput = styled.div`
    display: flex;
    flex-flow: row wrap;
    margin-bottom: 1em;
    border-bottom: solid 1px #04d092;

    div {
        -webkit-box-sizing: border-box;
        -moz-box-sizing: border-box;
        box-sizing: border-box;
        display: flex;
        flex-flow: row wrap;
        padding: 0.1em;
        margin-top: 0.4em;
        margin-bottom: 0.4em;      
        width: 100%;  

        label {
            min-width: 25%;
            padding: 0.2em;

        }

        input {
            margin-right: 5em;
            padding: 0.5em;
            width: 100%;

            @media (min-width: 800px) {
                width: 100%;
            }     
        }
    }
`;
export const Company = styled.div`
    display: flex;
    flex-flow: row wrap;

    div {
        -webkit-box-sizing: border-box;
        -moz-box-sizing: border-box;
        box-sizing: border-box;
        display: flex;
        flex-flow: row wrap;
        margin-top: 0.4em;
        margin-bottom: 0.4em;
        width:100%;

        input {
            width: 100%;

            @media (min-width: 800px) {
                width: 100%;
            }
        } 

        @media (min-width: 800px) {        
            width: 50%;
        }
    }
`;
export const Calendar = styled.div`
    display: flex;
    flex-flow: row no-wrap;
    width: 100%;
    
    div {
        -webkit-box-sizing: border-box;
        -moz-box-sizing: border-box;
        box-sizing: border-box;
        display: flex;
        flex-flow: column no-wrap;
        margin-top: 0.4em;
        margin-bottom: 0.4em;
        width: 100%;

        @media (min-width: 800px) {
            width: 50%;
        }

        label {
            width: 100%

            @media (min-width: 800px) {
                width: 75%;
        
            }
        } 
        div {
            display: flex;
            flex-flow: row no-wrap;
            width:100%;
            
            select {
                padding:0.1em;
                margin-right: 0.2em;
                font-size: 1.1em;
                width: 38%;
            }
        }
    }
`;
export const Experiences = styled(Websites)``;
export const XPTag = styled.div`
    width: 100%;
    display:flex;
    flex-flow: row wrap;   
    padding-left: 1em;
    padding-right: 1em;
    padding-top: 1em;
    padding-bottom: 1em;
    border-top: solid 1px #DDEDF5;

`;

export const Dates = styled.div`
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    display: flex;
    flex-flow: column wrap;
    width:100%;

    @media (min-width: 800px) { 

        width: 35%;
    }
`;
export const WorkDetails = styled.div`
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    display: flex;
    flex-flow: column wrap;
    width:100%;

    @media (min-width: 800px) { 
       
        width: 65%;
    }
`;
export const SinceDate = styled.p`
    margin: 0;
    padding-bottom: 0.4em;
    width: 100%;
    color: #1c75bc;  
`;
export const YearsMonths = styled(SinceDate)`
    color: #9b9b9b;
`;

export const CompanyLocation = styled.p`
    color: #9b9b9b;
    margin: 0;
    padding: 0;
    width: 100%;
`;
export const PostitionInCorp = styled.p`
    margin: 0;
    padding-top: 0.7em;
    width: 100%;
    font-size: 1.4em;
`;
export const Info = styled.p`
    margin: 0;
    padding-top: 0.8em;
    width: 100%;
`;


/* Start of Education Section styling */
export const EducationHead = styled(ExperienceHead)``;
export const EducationInput = styled(ExperienceInput)``;
export const Click2AddEdu = styled(Click2AddWork)``;
export const EduTag = styled(XPTag)``;
export const EduDetails = styled(WorkDetails)``;
export const SchoolLocation = styled(CompanyLocation)``;
export const Degree = styled(PostitionInCorp)``;
export const Schools = styled(Experiences)``;
export const WhatDegree = styled(Company)``;
export const WhatUni = styled(Company)``;

/* Start of About Me Section styling */
export const AboutMeHead = styled(ExperienceHead)``;
export const AboutMeBody = styled.div`
    display: flex;
    flex-flow: column no-wrap;
    padding-left: 1em;
    padding-right: 1em;
    padding-bottom: 1em;

    div {
        font-size: 0.95em;
        
        span {
            color: #9b9b9b;
            padding-left: 0.8em;
        }
    }
    
    p:not(:last-child) {
        font-size: 1.5em;
    }
`;
export const AboutMeInput = styled(ExperienceInput)`
    border-bottom: none;
`;
/* End of About Me Section styling */
