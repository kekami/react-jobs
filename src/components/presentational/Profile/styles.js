import styled from 'styled-components';
import { globalStyles } from '../globalStyles.js';

export const ProfileWrapper = styled.div`
    min-height: 100vh;
    min-width: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding-left: 5vw;
    padding-right: 5vw;
    padding-top: 10px;
    padding-bottom: 10px;
    background-color: #E1D6D6;
`;

export const ProfileContainer = styled.div`
 background-color: #fff;
 border-radius: 10px;
`;

export const ProfileNav = styled.div``;
export const ProgressBar = styled.div`
   
`;
export const SectionBox = styled.div`
     padding: 2em;
`;
export const SectionHead = styled.div`
    padding: 1em;
    
`;
export const Description = styled.div`
    color: grey;
`;
export const Link = styled.a``;
export const SectionWrap = styled.div``;

