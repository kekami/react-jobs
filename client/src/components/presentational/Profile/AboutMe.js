import React, { Component } from 'react';
import { SectionWrapper, AboutMeHead, AboutMeBody, Edit, I, AboutMeInput, Company } from './styles';

/* In this section I had to create a few input fields and their corresponding output elements.
User should be able to add his name, surname, location and short introduction.

I used pattern from LinkSection and used refs and their corresponding fields to present user inputs.
All input fields deliver submitted change to state from which presentational component draws data.
*/

export class AboutMe extends Component {
  constructor(props) {
    super(props);

    this.state = {
      FirstName: '',
      LastName: '',
      Location: '',
      Intro: '',
      show: false,
    };

    this.onClick = this.onClick.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentWillMount() {
    window.scrollTo(0, 0);
  }

  onClick() {
    this.setState({ show: !this.state.show });
  }

  handleSubmit(e) {
    e.preventDefault();

    this.setState({
      FirstName: this.FirstName.value,
      LastName: this.LastName.value,
      Location: this.Location.value,
      Intro: this.Intro.value,
    });
  }


  render() {
    return (
      <SectionWrapper edited={this.state.show} style={{ backgroundColor: '#FAFAFA' }} >
        <AboutMeHead edited={this.state.show}>
          <p><strong>About me</strong></p>
          <Edit onClick={this.onClick} edited={this.state.show}>
            { this.state.show ?
              <div><I className="fa fa-times" aria-hidden="true" /></div>
              : <div>Edit <I className="fa fa-pencil" aria-hidden="true" /></div> }
          </Edit>
        </AboutMeHead>
        <AboutMeBody style={{ display: this.state.show ? 'none' : 'block' }}>
          <div>
            <div><strong>Name</strong> </div>
            <p id="names">{this.state.FirstName} {this.state.LastName}</p>
            <div><strong>Living in</strong></div>
            <p id="location">{this.state.Location}</p>
            <div><strong>Introduction </strong> <span>Just a few words about me</span> </div>
            <p id="introduction">{this.state.Intro}</p>
          </div>
        </AboutMeBody>
        <form action="" style={{ display: this.state.show ? 'block' : 'none' }} onSubmit={this.handleSubmit}>
          <AboutMeInput >
            <Company>
              <div className="form-group">
                <label htmlFor="firstName"><strong>First Name</strong>
                  <input
                    id="firstName"
                    type="text"
                    name="FirstName"
                    placeholder="e.g. John"
                    ref={(input) => { this.FirstName = input; }}
                  />
                </label>
              </div>
              <div className="form-group">
                <label htmlFor="lastName"><strong>Last name</strong>
                  <input
                    id="lastName"
                    type="text"
                    name="LastName"
                    placeholder="e.g. Galt"
                    ref={(input) => { this.LastName = input; }}
                  />
                </label>
              </div>
            </Company>
            <div className="form-group">
              <label htmlFor="location"><strong>Living in</strong>
                <input
                  id="location"
                  type="text"
                  name="location"
                  placeholder="e.g. Warsaw, Poland"
                  ref={(input) => { this.Location = input; }}
                />
              </label>
            </div>
            <div className="form-group">
              <label htmlFor="introduction"><strong>Introduction just a few words about me</strong>
                <input
                  id="introduction"
                  type="text"
                  name="intro"
                  ref={(input) => { this.Intro = input; }}
                />
              </label>
            </div>
          </AboutMeInput>
          <button className="btn btn-primary" type="submit">Save</button>
        </form>
      </SectionWrapper>
    );
  }
}
