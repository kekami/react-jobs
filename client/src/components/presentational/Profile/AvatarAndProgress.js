import React, { Component } from 'react';
import { UpperSectionWrapper, PicWrapper, Avatar, PicUploader, ProgressSection, ProgressWrapper, ProgressBar } from './styles';

/* In this section I learnt how to dynamically change styles
 and pass them to the target component with the use of props utilising template:
-->> '${<enter referenced js variable>}<here add anything to concatenate the string you create' */

/* To make sure that HTML input/ browse button will fill parent's space:
- parent has to have position: relative;
- <input/> needs: position:absolute with top:0, left:0 (X,Y axis) and width&height of parent or 100%


Although small, this component taught me a lot not only functionality-wise but also about styling
*/
export class AvatarAndProgress extends Component {
  constructor(props) {
    super(props);

    this.state = {
      width100: 10,
      // here you will find an implementation of technical background behind % bar
    };

    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    const width100 = this.state.width100;
    if (width100 < 100) {
      this.setState({
        width100: (width100 + 10),
      });
    } else {
      this.setState({
        width100: 10,
      });
    }
  }

  move() {
    const width100 = this.state.width100;
    return {
      width: `${width100}%`,
    };
  }


  render() {
    return (
      <UpperSectionWrapper>
        <PicWrapper>
          <Avatar className="avatar" src="/img/avatar.png" />
        </PicWrapper>
        <div>
          <PicUploader >
            <div>
              <input type="file" accept="image/png,image/jpeg" />
              Click here to upload a png or jpeg file here (max 10 MB).
              This picture will be visible to employers, so choose one
              that is a good representation of you!
            </div>
          </PicUploader>
        </div>
        <ProgressSection>
          <header>Your profile is <strong>{this.state.width100}%</strong> complete</header>
          <ProgressWrapper>
            <ProgressBar className="progressBar" onClick={this.onClick} style={this.move()} />
          </ProgressWrapper>
        </ProgressSection>
        <div>
          <p>This is your profile that will represent you when you apply for jobs on Jobbatical.
            By submitting an application, you will make your profile visible
            to the company youâ€™re applying to.
            Note that changes made here will also affect previously submitted applications.</p>
        </div>
      </UpperSectionWrapper>
    );
  }
}
