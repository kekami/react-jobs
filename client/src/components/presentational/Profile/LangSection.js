import React, { Component } from 'react';
import { SectionWrapper, First2, Edit, Btn, Inputform, I, TagsForm, LangTag, LangHead } from './styles';

/* In section of Languages my goal was to create similar tags to those found in Skill Section.
Here is recap of the goals from SkillSection.js:
I wanted to make sure user can easily and intuitively add and remove any tag.
1. Every tag had to be generated based on user input and visibile right after submission (no delay).
2. The input field should stay hidden until user decides to add new tags.

I learnt how to use state (inputValue) of this component which updates on change of the input field.
Upon submission (enter) a function handling submit creates a new tag with submitted input value.
This tag is later added to the array [tags] stored in state and at the end inputValue is cleared.

Tags are added by mapping over tags array. Each tag is equipped with deleteTag() onClick handler.
This enables user to click on the tag (with visible 'x' button on it) and delete unwanted elements.
The deleteTag() function removes designated tag from the array after click and distinguishes between
them thanks to index number of every tag - it leaves tags which were not clicked untouched.
*/

export class LangSection extends Component {
  constructor(props) {
    super(props);

    this.state = {
      show: false,
      tags: [],
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.onClick = this.onClick.bind(this);
    this.deleteTag = this.deleteTag.bind(this);
  }

  onClick() {
    this.setState({ show: !this.state.show });
  }
  deleteTag(index) {
    const tags = this.state.tags.filter((tag, i) => i !== index);
    this.setState({ tags });
  }

  handleSubmit(event) {
    event.preventDefault();
    const newTag = {
      title: this.state.inputValue,
      isShowing: true,
    };
    const tags = this.state.tags.concat(newTag);
    this.setState({
      tags,
      inputValue: '',
    });
  }

  addingTags() {
    return this.state.tags.map((tag, i) => (
      <LangTag key={tag.title} onClick={() => this.deleteTag(i)} showing={tag.isShowing}>
        <Btn>{tag.title} <I className="fa fa-times" aria-hidden="true" /></Btn>
      </LangTag>
    ),
    );
  }

  render() {
    return (
      <SectionWrapper edited={this.state.show}>
        <First2>
          <LangHead edited={this.state.show}>
            <p><strong>Languages</strong> My languages</p>
            <Edit onClick={this.onClick} edited={this.state.show}>
              { this.state.show ?
                <div><I className="fa fa-times" aria-hidden="true" /></div>
                : <div>Edit <I className="fa fa-pencil" aria-hidden="true" /></div> }

            </Edit>
          </LangHead>
          <Inputform style={{ display: this.state.show ? 'block' : 'none' }} onSubmit={this.handleSubmit}>
            <input
              type="text"
              onChange={e => this.setState({ inputValue: e.target.value })}
              value={this.state.inputValue}
              placeholder="Type in a skill and hit Enter"
            />
          </Inputform>
          <TagsForm>
            {this.addingTags()}
          </TagsForm>
        </First2 >
      </SectionWrapper>
    );
  }
}
