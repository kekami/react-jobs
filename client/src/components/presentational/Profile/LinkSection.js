import React, { Component } from 'react';
import { SectionWrapper, LinkHead, Edit, I, SocMediaInput, LinkTag, Websites } from './styles';

/* 
Using knowledge from Skill and Lang sections I endeavoured to generate more predictable LinkSection.
My goal here was to allow user to populate this section with his social media links.
1. User have to be able to add/edit his links - 5 input fields.
2. Unused input fields should not generate empty outputs - only added links should be displayed.

Since this section allowed only for 5 inputs I decided to make an array of 5 objects.
Each object consists of index, logo and URL. URL is a value of referenced input. 
This makes sure all inputs always find their own place and logo.
Lastly, submitData() makes sure to filter out from this array objects without url.

Just like in previous sections addingLinks method maps over the Links[] and outputs user links.

I learnt here how to use refs and that that I can store whole divs inside an object.

*/

export class LinkSection extends Component {
  constructor(props) {
    super(props);

    this.state = {
      show: false,
      Links: [],
    };
    this.onClick = this.onClick.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  onClick() {
    this.setState({ show: !this.state.show });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.submitData();
  }

  submitData() {
    let Links = this.state.Links;
    Links = [
      { index: 1,
        logo: <div><i className="fa fa-home" aria-hidden="true" /></div>,
        url: this.website.value },
      { index: 2,
        logo: <div><i className="fa fa-linkedin-square" aria-hidden="true" /></div>,
        url: this.linkedIn.value },
      { index: 3,
        logo: <div><i className="fa fa-facebook-square" aria-hidden="true" /></div>,
        url: this.facebook.value },
      { index: 4,
        logo: <div><i className="fa fa-twitter-square" aria-hidden="true" /></div>,
        url: this.twitter.value },
      { index: 5,
        logo: <div><i logo="g+" className="fa fa-google-plus-square" aria-hidden="true" /></div>,
        url: this.googlepl.value },
    ];
    Links = Links.filter(link => (link.logo && link.url.length));
    this.setState({ Links });
  }

  addingLinks() {
    return this.state.Links.map(link => (
      <LinkTag key={link.index} showingbelow="true">
        {link.logo} <p>{link.url}</p>
      </LinkTag>
    ));
  }
  render() {
    return (
      <SectionWrapper edited={this.state.show}>
        <LinkHead edited={this.state.show}>
          <p><strong>Links</strong> My social media and homepage links</p>
          <Edit onClick={this.onClick} edited={this.state.show}>
            { this.state.show ?
              <div><I className="fa fa-times" aria-hidden="true" /></div>
              : <div>Edit <I className="fa fa-pencil" aria-hidden="true" /></div> }
          </Edit>
        </LinkHead>
        <form action="" style={{ display: this.state.show ? 'block' : 'none' }} onSubmit={this.handleSubmit}>
          <p>
            Add some extra firepower to your profile by providing
            links to professional and social media profiles.
          </p>
          <SocMediaInput>
            <div className="form-group">
              <label htmlFor="website">Your website </label>
              <input
                id="website"
                name="website"
                ref={(input) => { this.website = input; }}
                type="url"
                placeholder="paste your link here and hit Enter"
              />
            </div>
            <div className="form-group">
              <label htmlFor="linkedIn">Linkedin profile </label>
              <input
                id="linkedIn"
                name="linkedIn"
                ref={(input) => { this.linkedIn = input; }}
                type="url"
                placeholder="paste your link here and hit Enter"
              />
            </div>
            <div className="form-group">
              <label htmlFor="facebook">Facebook profile </label>
              <input
                id="facebook"
                name="Facebook"
                ref={(input) => { this.facebook = input; }}
                type="url"
                placeholder="paste your link here and hit Enter"
              />
            </div>
            <div className="form-group">
              <label htmlFor="twitter">Twitter profile </label>
              <input
                id="twitter"
                name="Twitter"
                ref={(input) => { this.twitter = input; }}
                type="url"
                placeholder="paste your link here and hit Enter"
              />
            </div>
            <div className="form-group">
              <label htmlFor="googlePlus">Google+ profile </label>
              <input
                id="googlePlus"
                name="google+"
                ref={(input) => { this.googlepl = input; }}
                type="url"
                placeholder="paste your link here and hit Enter"
              />
            </div>
          </SocMediaInput>
          <button className="btn btn-primary" type="submit">Save</button>
        </form>
        <Websites className="addedLinks" style={{ display: !this.state.show ? 'block' : 'none' }}>
          {this.addingLinks()}
        </Websites>
      </SectionWrapper>
    );
  }
}
