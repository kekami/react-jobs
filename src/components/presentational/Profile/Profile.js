import React from 'react';
// import { Link } from 'react-router-dom';
import { ProfileWrapper, ProfileContainer, ProfileNav, ProgressBar, SectionBox, SectionHead, Link, SectionWrap, Description } from './styles.js';

const sections = [
  { Title: 'About', Desc: 'About you' },
  { Title: 'Skills', Desc: 'diff skills' },
  { Title: 'Language skills', Desc: 'My languages & proficiency levels' },
  { Title: 'Links', Desc: 'My social media & homepage links' },
  { Title: 'Work Experience', Desc: 'Previous & current positions' },
  { Title: 'Education', Desc: 'Schools & courses' },
];


export function Profile() {
  return (
    <ProfileWrapper>
      <ProfileContainer>
        <ProfileNav>
          <ul>
            <li><Link to="/">My Profile & CV</Link></li>
            <li><Link to="/">My Preferences</Link></li>
          </ul>
        </ProfileNav>
        <ProgressBar>
          <span>Here we will drop avatar and lift it up to cover space in between of 2 tabs</span>
        </ProgressBar>

        <ProfileSections sections={sections} />
      </ProfileContainer>
    </ProfileWrapper>
  );
}

const ProfileSections = ({ sections }) => {
  const sectionsJSX = sections.map(section => (
    <SectionBox>
      <SectionHead>
        <strong>{section.Title}</strong>
        <Description>
          {section.Desc}
        </Description>
      </SectionHead>
    </SectionBox>
  ));

  return (
    <SectionWrap>
      {sectionsJSX}
    </SectionWrap>
  );
};
