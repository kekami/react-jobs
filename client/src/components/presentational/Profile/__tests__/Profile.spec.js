import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import { Profile } from '../Profile';

// Tests for Profile Section
/*
1. basic render
2. snapshot
*/

describe('Profile component - basic test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Profile />
      </MemoryRouter>, div);
  });

  it('Profile component matches its shallow snapshot', () => {
    const wrapper = shallow(<Profile />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
