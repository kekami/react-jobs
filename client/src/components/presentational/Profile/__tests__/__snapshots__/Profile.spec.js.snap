// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component - basic test Profile component matches its shallow snapshot 1`] = `
<styled.div>
  <styled.div>
    <ProfileNav />
    <AvatarAndProgress />
    <AboutMe />
    <SkillSection />
    <LangSection />
    <LinkSection />
    <WorkExperience />
    <Education />
  </styled.div>
</styled.div>
`;
